src/masonite/cache/drivers/MemcacheDriver.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+
+class MemcacheDriver:
+    def __init__(self, application):
+        self.application = application
+        self.connection = None
+
+    def set_options(self, options):
+        self.options = options
+        return self
+
+    def get_connection(self):
+        try:
+            from pymemcache.client.base import Client
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not find the 'pymemcache' library. Run 'pip install pymemcache' to fix this."
+            )
+
+        if not self.connection:
+            if str(self.options.get("port")) != "0":
+                self.connection = Client(
+                    f"{self.options.get('host')}:{self.options.get('port')}"
+                )
+            else:
+                self.connection = Client(f"{self.options.get('host')}")
+
+        return self.connection
+
+    def add(self, key, value):
+        if self.has(key):
+            return self.get(key)
+
+        self.put(key, value)
+        return value
+
+    def get(self, key, default=None, **options):
+        if not self.has(key):
+            return default
+
+        return self.get_value(
+            self.get_connection().get(f"{self.get_name()}_cache_{key}")
+        )
+
+    def put(self, key, value, seconds=0, **options):
+        if isinstance(value, (dict, list)):
+            value = json.dumps(value)
+
+        return self.get_connection().set(
+            f"{self.get_name()}_cache_{key}", value, expire=seconds
```

Comment:
```python
f"{self.get_name()}_cache_{key}"
```
It's a convention you choose for the key ? To provide "unicity" ? (I mean instead of just using `key`

Reply:
Took it from the redis integration someone did for Masonite .. I think they did it because if you have multiple apps using the same memcache /redis server